PolyEngine/Engine/Src/Engine.hpp

```diff
@@ -22,10 +23,17 @@ namespace Poly {
 
 		void Update(float dt);
 
+		void KeyDown(eKey key) { InputBase.Push({eEventType::KEYDOWN, key}); }
+		void KeyUp(eKey key) { InputBase.Push({eEventType::KEYUP, key}); }
+		void UpdateMousePos(Vector pos) { InputBase.Push({eEventType::MOUSEMOVE, pos}); }
+		void UpdateWheelPos(Vector pos) { InputBase.Push({eEventType::WHEELMOVE, pos}); }
```

Comment:
Please pass Vector, and other structs/classes by const reference instead of value. This is optimization.